sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -686,6 +686,189 @@ case class EndsWith(left: Expression, right: Expression) extends StringPredicate
     newLeft: Expression, newRight: Expression): EndsWith = copy(left = newLeft, right = newRight)
 }
 
+/**
+ * A function that checks if a UTF8 string is valid.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(str) - Returns true if `str` is a valid UTF-8 string, otherwise returns false.",
+  arguments = """
+    Arguments:
+      * str - a string expression
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_('Spark');
+       true
+      > SELECT _FUNC_(x'61');
+       true
+      > SELECT _FUNC_(x'80');
+       false
+      > SELECT _FUNC_(x'61C262');
+       false
+  """,
+  since = "4.0.0",
+  group = "string_funcs")
+case class IsValidUTF8(input: Expression) extends RuntimeReplaceable with ImplicitCastInputTypes
```

Comment:
Do we want this expression to be ImplicitCastInputTypes or ExpectsInputType only? Difference is that latter would only fail on getting different input.

Reply:
I don't have a strong opinion. On one hand, it looks weird to allow int/double inputs for `IsValidUTF8`, on the other hand, it's simpler to have the same implicit cast rule for all functions.

cc @srielau @dtenedor

Reply:
update: consulted with @srielau and decided to keep `ImplicitCastInputTypes` here

it's how most string functions work in Spark, and we shouldn't make any exceptions for the new expressions